### ISSUE TITLE
Validation missing for column type definition in createTable

### ISSUE THREAD

**Issue by hemidactylus** [stargate/data-api#1560]
(tested with `stargateio/data-api:v1.0.18` locally)

One can pass any additional field to the column definition in createTable and no validation errors are received, e.g.

```
"columns": [
  {"type": "int", "favorite_color": "teal"}
]
```

**Comment by tatu-at-datastax** [comment]
I cannot reproduce and we have an integration for this case (but I wrote another one locally just to be sure).

@hemidactylus I'll close this, presumably fixed; if you can still reproduce with 1.0.19-SNAPSHOT please re-open and include full request used?

**Comment by hemidactylus** [comment]
I can reproduce with the latest main as of now (tag v1.0.20), both on locally-running and against Astra DEV. Complete (successful) payload:

```
{
    "createTable": {
        "name": "table_altered",
        "definition": {
            "columns": {
                "p_text": {
                    "type": "text"
                },
                "p_int": {
                    "type": "int",
                    "favorite_color": "teal"
                },
                "p_vector": {
                    "type": "vector",
                    "dimension": 1024
                }
            },
            "primaryKey": "p_text"
        }
    }
}
```

**Comment by hemidactylus** [comment]
@tatu-at-datastax sorry but I don't have the permissions to re-open this one, I need to ask you to do it for me.

**Comment by tatu-at-datastax** [comment]
Ahhhh! Different from original reproduction: not "simple" bad type bad Object with extra properties. Not sure if that's bug or not... need to think.

**Comment by tatu-at-datastax** [comment]
@maheshrajamani WDYT? The issue is that extra properties like this one:

```
                "p_int": {
                    "type": "int",
                    "favorite_color": "teal"
                },
```

are not considered errors -- my initial reaction is that I agree with @hemidactylus and they should be, but I think you have most context about this as to whether it's a thing to fix, or one to document.

**Comment by vkarpov15** [comment]
My 2 cents: I think it is reasonable to consider extra properties an error. In Mongoose and app dev work I typically ignore extraneous properties rather than erroring, but that is because for Mongoose plugins may use the extraneous properties, and for app dev extraneous properties are typically not a big deal. However, a misconfigured collection due to a property typo is a difficult mistake to fix, so I think erroring is reasonable here.

**Comment by maheshrajamani** [comment]
> @maheshrajamani WDYT? The issue is that extra properties like this one:

I think it's a bug. If extra/unsupported properties are provided the api need to error out.

**Comment by tatu-at-datastax** [comment]
Ok. We can go with that.

I suspect we have other places where we do not check for extra properties, but only for mandatory and optional ones. But they can be handled while we come across (via client devs reporting or when we change code in relevant areas).

**Comment by amorton** [comment]
yeah prob a non trivial as we should check how we handle un expected fields in all places, this will be complicated by the use of Custom deserializers.
